pkg/dial/resolver.go

```diff
@@ -28,31 +28,40 @@ import (
 )
 
 // ServiceResolverCell provides a ServiceResolver instance to map DNS names
-// matching Kubernetes services to the corresponding ClusterIP address, backed
-// by a lazy resource.Store, which is only started on first access.
+// matching Kubernetes services to the corresponding ClusterIP address
+// using the LB frontends table.
 var ServiceResolverCell = cell.Module(
 	"service-resolver",
 	"Service DNS names to ClusterIP translator",
 
-	cell.Provide(newServiceResolver),
+	cell.Provide(newLBServiceResolver),
 )
 
-var _ Resolver = (*ServiceResolver)(nil)
+// ResourceServiceResolverCell provides a ServiceResolver instance to map DNS names
+// matching Kubernetes services to the corresponding ClusterIP address using the
+// services resource. Used by the operator.
+var ResourceServiceResolverCell = cell.Module(
```

Comment:
Does something prevents us from accidentally using `ResourceServiceResolverCell` in Cilium Agent and starting additional informers?